Sources/Sentry/SentryDependencyContainer.m

```diff
@@ -54,6 +53,24 @@
 #    import "SentryReachability.h"
 #endif // !TARGET_OS_WATCH
 
+/**
+ * Macro for implementing lazy initialization with a double-checked lock. The double-checked lock
+ * speeds up the dependency retrieval by around 5%, so it's worth having it. Measured with
+ * self.measure in unit tests.
+ */
+#define SENTRY_LAZY_INIT(instance, initBlock)                                                      \
+    if (instance == nil) {                                                                         \
+        @synchronized(sentryDependencyContainerDependenciesLock) {                                 \
+            if (instance == nil) {                                                                 \
+                instance = initBlock;                                                              \
+            }                                                                                      \
+        }                                                                                          \
+    }                                                                                              \
+    return instance;
```

Comment:
Returning inside the macro will make things trickier, for instance, dealing with the inout error parameter in the crashing file manager property init.

Reply:
Proposed the change in https://github.com/getsentry/sentry-cocoa/pull/5246

Reply:
Thanks for the proposal. I found a different approach that doesn't require moving returning the instance out of the macro https://github.com/getsentry/sentry-cocoa/blob/157c2d3301c5510a1e66e90d42d4a236347e87cf/Sources/Sentry/SentryDependencyContainer.m#L172-L183

I think this approach is also legit. Thanks for opening a PR for the suggestions.